operator/cmd/mesh/upgrade.go

```diff
@@ -152,22 +153,22 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l *Logger) (err error) {
 	}
 	l.logAndPrintf("Upgrade version check passed: %v -> %v.\n", currentVersion, targetVersion)
 
-	// Read the overridden IOPS from args.inFilename
+	// Read the overridden IOPS from args.inFilenames
 	overrideIOPSYaml := ""
-	if args.inFilename != nil {
-		overrideIOPSYaml, err = ReadLayeredYAMLs(args.inFilename)
+	if args.inFilenames != nil {
+		overrideIOPSYaml, err = ReadLayeredYAMLs(args.inFilenames)
 		if err != nil {
-			return fmt.Errorf("failed to read override IOPS from file: %v, error: %v", args.inFilename, err)
+			return fmt.Errorf("failed to read override IOPS from file: %v, error: %v", args.inFilenames, err)
 		}
 	}
 
-	// Generates IOPS for args.inFilename IOP specs yaml. Param force is set to true to
+	// Generates IOPS for args.inFilenames IOP specs yaml. Param force is set to true to
 	// skip the validation because the code only has the validation proto for the
 	// target version.
-	currentIOPSYaml, _, err := genIOPS(args.inFilename, "", "", currentVersion, true, kubeClient.Config, l)
+	currentIOPSYaml, _, err := GenerateConfig(args.inFilenames, "", true, nil, l)
```

Comment:
How does GenerateConfig know the current version? I think this will break upgrade check. We will not be able to compare the default value between versions.

Reply:
Why does GenerateConfig need to have version passed in? Version concept is not used in the code itself. The only control you have is which charts/values you pass into this function.